src/sidebar/services/annotations.js

```diff
@@ -12,6 +17,80 @@ export default function annotationsService(
 ) {
   let searchClient = null;
 
+  /**
+   * Extend new annotation objects with defaults and permissions.
+   */
+  function initialize(annotationData, now = new Date()) {
+    const defaultPrivacy = store.getDefault('annotationPrivacy');
+    const groupid = store.focusedGroupId();
+    const profile = store.profile();
+
+    const userid = profile.userid;
+    const userInfo = profile.user_info;
+
+    let permissions = defaultPermissions(userid, groupid, defaultPrivacy);
+
+    // Highlights are peculiar in that they always have private permissions
+    if (metadata.isHighlight(annotationData)) {
+      permissions = privatePermissions(userid);
+    }
+
+    return Object.assign(
+      {
+        created: now.toISOString(),
+        group: groupid,
+        permissions,
+        tags: [],
+        text: '',
+        updated: now.toISOString(),
+        user: userid,
+        user_info: userInfo,
+        $tag: window.btoa(Math.random()),
+      },
+      annotationData
+    );
+  }
+
+  /**
+   * Populate a new annotation object from `annotation` and add it to the store.
+   * Create a draft for it unless it's a highlight and clear other empty
+   * drafts out of the way.
+   */
+  function create(annotationData, now = new Date()) {
+    const annotation = initialize(annotationData, now);
+
+    // Remove other drafts that are in the way, and their annotations (if new)
+    store.deleteNewAndEmptyDrafts();
```

Comment:
Aside: `deleteNewAndEmptyDrafts` is also a candidate for conversion to a `service` method. It's a little funky at present.